### ISSUE TITLE
Old Query Param Value Gets Rehydrated After Navigation

### ISSUE THREAD

**Issue by tariwiencke** [47ng/nuqs#774]
## Context

What's your version of `nuqs`?

```
"nuqs": "^2.2.1",
```

What framework are you using?

- ✅ Next.js (app router)

Which version of your framework are you using?

```
"next": "15.0.3", "react": "19.0.0-rc-66855b96-20241106",
```

## Description

<img width="1061" alt="Screenshot 2024-11-18 at 22 55 23" src="https://github.com/user-attachments/assets/e372fd14-ae4f-4339-a23f-9f6bf58cfd2a">

When navigating pages with the link component the value of url state gets reset in url but when changing a different value of query params object the old value gets rehydrated.
## Reproduction

CodeSandbox: https://codesandbox.io/p/github/tariwiencke/nuqs-bug/main?import=true

1. Change values of both inputs
2. Navigate with link component `Go to Subpage`
3. Change value of one input
4. Bug: old value of second input gets rehydrated and is not displayed in url

**Comment by franky47** [comment]
Thanks for the report, I managed to reproduce it on the e2e test bench too, it looks like it's an issue only in useQueryStates: splitting the search params into individual useQueryState hooks doesn't show this behaviour.

I'll have a look and report back.

**Comment by franky47** [comment]
I think I have a fx in #776, could you try this and let me know if it fixes the issue on your end please?

```
pnpm add https://pkg.pr.new/nuqs@776
```

**Comment by tariwiencke** [comment]
That works just fine. Thanks.

**Comment by franky47** [comment]
This was shipped in 2.2.2.
